error message - Function 'last_value(`t1_86`.`c_1_2`)' not supported with OVER clause

--java.lang.AssertionError: WITH with_t_0 as (SELECT t1_86.c_1_0, t1_86.c_1_1 FROM t1 AS t1_86) , with_t_1 as (SELECT t1_86.c_1_0, t1_86.c_1_1 FROM t1 AS t1_86) SELECT t1_86.c_1_0, LAST_VALUE(t1_86.c_1_2) OVER ( PARTITION BY t1_86.c_1_2)  FROM t1 AS t1_86, with_t_0 AS with_t_0, with_t_1 AS with_t_1 WHERE false GROUP BY t1_86.c_1_2;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:128)
--	at sqlancer.ComparatorHelper.getResultSetFirstColumnAsString(ComparatorHelper.java:54)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPHavingOracle.check(StarRocksTLPHavingOracle.java:38)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:322)
--	at sqlancer.Main$2.run(Main.java:511)
--	at sqlancer.Main$2.runThread(Main.java:489)
--	at sqlancer.Main$2.run(Main.java:479)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Function 'last_value(`t1_86`.`c_1_2`)' not supported with OVER clause.
--	at sun.reflect.GeneratedConstructorAccessor5.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:118)

### Steps to reproduce the behavior (Required)
CREATE DATABASE h8;
USE h8;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL ,c_0_1 DATETIME NOT NULL ,c_0_2 DATE NULL ,c_0_3 DATE NOT NULL ,c_0_4 SMALLINT NULL ,c_0_5 BOOLEAN NULL ,c_0_6 BOOLEAN NOT NULL ,c_0_7 BOOLEAN NOT NULL ,c_0_8 BOOLEAN NULL ,c_0_9 ARRAY<STRING> NULL ,c_0_10 DATETIME NOT NULL ,c_0_11 ARRAY<STRING> NOT NULL ,c_0_12 FLOAT NULL ,c_0_13 ARRAY<INT> NOT NULL ,c_0_14 BOOLEAN NOT NULL ,c_0_15 INT NOT NULL ,c_0_16 DATE NOT NULL , INDEX index1_c_0_6(`c_0_6`) USING BITMAP) UNIQUE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8) PARTITION BY RANGE(c_0_2) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 219 day)) DISTRIBUTED BY HASH (c_0_5,c_0_2) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_0_4") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 BOOLEAN NULL ,c_1_1 VARCHAR(1) NULL ,c_1_2 ARRAY<INT> NOT NULL ) UNIQUE KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 VARCHAR(51) NOT NULL ,c_2_1 DOUBLE MAX NULL ,c_2_2 VARCHAR(21) MAX NOT NULL ,c_2_3 FLOAT REPLACE_IF_NOT_NULL NULL ,c_2_4 BOOLEAN REPLACE NOT NULL ) AGGREGATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE MATERIALIZED VIEW v0 AS SELECT t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4, MAX(t2_78.c_2_1) , MAX(t2_78.c_2_2)  FROM t2 AS t2_78 GROUP BY t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4 ORDER BY t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4 DESC;
CREATE MATERIALIZED VIEW v0 AS SELECT t1_80.c_1_1, MIN(t1_80.c_1_0)  FROM t1 AS t1_80 GROUP BY t1_80.c_1_1 ORDER BY t1_80.c_1_1 ASC;
INSERT INTO t0 VALUES ("2010-01-03 08:00:14", "2010-01-07 08:00:14", "2010-01-23", "2010-01-23", 3, false, false, true, true, ["", "bzS", "GDkb2MWdZ", "b", "cd", "", "zQwgi", "iOYkX6tz", "Ur0", "", "K2zZdIoS", "mJhe542TI"], "2010-01-17 08:00:14", ["", ""], 0.3733675486428276, [412985474, -99331877, -1141428404, 412985474, -99331877, -1141428404, 412985474, 1529649165, 830533870, 1529649165, -17167736], false, 12, "2010-01-06"), ("2010-01-23 08:00:14", "2010-01-16 08:00:14", "2010-01-23", "2010-01-19", NULL, false, true, false, false, ["frXOu400M", "2EsS", "u4", "L", "X", "zW", "L"], "2010-01-27 08:00:14", ["", "dgfKI", "f0RMp5bj"], 0.1848353313517348, [-2081342657, 1518150261, 1228979431, -994881310, -282219595], false, 14, "2010-01-05");
INSERT INTO t2 VALUES ("ww", 0.3974359266304929, "l", 0.18119365268821286, false), ("", 0.22603276267429873, "ItRqdiMy", 8.3053387E8, false), ("", 0.18440813391504096, "D", 0.22603276267429873, true);
INSERT INTO t2 VALUES ("F9", 0.2055758386797617, "pn", 0.6448639098327671, false), ("UcBYW", 0.2055758386797617, "", 0.18119365268821286, false);

### Expected behavior query with last_value window function executed ok.

### Real behavior Function 'last_value(`t1_86`.`c_1_2`)' not supported with OVER clause.

### StarRocks version be05a2c20b9589ec947268b7f78860fb6fabbd46

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
@@ -815,6 +815,8 @@ public class FunctionSet {
                 Type.VARCHAR, Type.VARCHAR, true, false, false));
 
         for (Type t : Type.getSupportedTypes()) {
+            // null/char/time is handled through type promotion
+            // We do not support array/json/pseudo type now
             if (t.isNull() || t.isChar() || t.isTime() || t.isArrayType() || t.isJsonType() || t.isPseudoType()) {
                 continue;
             }

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
@@ -816,7 +816,7 @@ public class FunctionSet {
 
         for (Type t : Type.getSupportedTypes()) {
             // null/char/time is handled through type promotion
-            // array/json/pseudo is not supported yet
+            // TODO: array/json/pseudo is not supported yet
             if (t.isNull() || t.isChar() || t.isTime() || t.isArrayType() || t.isJsonType() || t.isPseudoType()) {
                 continue;
             }

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
@@ -815,13 +815,7 @@ public class FunctionSet {
                 Type.VARCHAR, Type.VARCHAR, true, false, false));
 
         for (Type t : Type.getSupportedTypes()) {
-            if (t.isNull()) {
-                continue; // NULL is handled through type promotion.
-            }
-            if (t.isChar()) {
-                continue; // promoted to STRING
-            }
-            if (t.isTime()) {
+            if (t.isNull() || t.isChar() || t.isTime() || t.isArrayType() || t.isJsonType() || t.isPseudoType()) {
                 continue;
             }
             addBuiltin(AggregateFunction.createAnalyticBuiltin(

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java
@@ -816,7 +816,7 @@ public class FunctionSet {
 
         for (Type t : Type.getSupportedTypes()) {
             // null/char/time is handled through type promotion
-            // We do not support array/json/pseudo type now
+            // array/json/pseudo is not supported yet
             if (t.isNull() || t.isChar() || t.isTime() || t.isArrayType() || t.isJsonType() || t.isPseudoType()) {
                 continue;
             }

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java
@@ -190,19 +190,13 @@ public class ArrayTypeTest extends PlanTestBase {
     }
 
     @Test
-    public void testArrayWindowFunction() {
+    public void testArrayWindowFunction() throws Exception {
         for (String fnName : Sets.newHashSet(AnalyticExpr.LASTVALUE, AnalyticExpr.FIRSTVALUE)) {
-            try {
-                String sql = String.format("select %s(v3) over() from tarray", fnName.toLowerCase());
-                getFragmentPlan(sql);
-            } catch (Exception e) {
-                e.printStackTrace();
-                assertContains(e.getMessage(),
-                        String.format("No matching function with signature: %s(ARRAY<bigint(20)>)",
-                                fnName.toLowerCase()));
-                continue;
-            }
-            throw new AssertionError();
+            String sql = String.format("select %s(v3) over() from tarray", fnName.toLowerCase());
+            expectedEx.expect(SemanticException.class);
+            expectedEx.expectMessage(
+                    String.format("No matching function with signature: %s(ARRAY<bigint(20)>)", fnName.toLowerCase()));
+            getFragmentPlan(sql);
         }
     }
 }

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java
@@ -2,6 +2,8 @@
 
 package com.starrocks.sql.plan;
 
+import com.google.common.collect.Sets;
+import com.starrocks.analysis.AnalyticExpr;
 import com.starrocks.sql.analyzer.SemanticException;
 import com.starrocks.utframe.StarRocksAssert;
 import org.junit.Assert;
@@ -185,5 +187,22 @@ public class ArrayTypeTest extends PlanTestBase {
         Assert.assertTrue(plan.contains("  8:Project\n" +
                 "  |  <slot 8> : array_contains(ARRAY<bigint(20)>[7: expr], 1)\n" +
                 "  |  <slot 9> : array_contains(ARRAY<bigint(20)>[7: expr], 2)"));
+    }
+
+    @Test
+    public void testArrayWindowFunction() {
+        for (String fnName : Sets.newHashSet(AnalyticExpr.LASTVALUE, AnalyticExpr.FIRSTVALUE)) {
+            try {
+                String sql = String.format("select %s(v3) over() from tarray", fnName.toLowerCase());
+                getFragmentPlan(sql);
+            } catch (Exception e) {
+                e.printStackTrace();
+                assertContains(e.getMessage(),
+                        String.format("No matching function with signature: %s(ARRAY<bigint(20)>)",
+                                fnName.toLowerCase()));
+                continue;
+            }
+            throw new AssertionError();
+        }
     }
 }

